ms2deepscore/MS2DeepScoreMonteCarlo.py

```diff
@@ -0,0 +1,171 @@
+from typing import List, Tuple
+import numpy as np
+from matchms import Spectrum
+from matchms.similarity.BaseSimilarity import BaseSimilarity
+from tqdm import tqdm
+
+from .vector_operations import cosine_similarity_matrix, mean_pooling, std_pooling
+from .typing import BinnedSpectrumType
+
+
+class MS2DeepScoreMonteCarlo(BaseSimilarity):
+    """Calculate MS2DeepScore ensemble similarity scores and STD between a reference
+    and a query using Monte-Carlo Dropout
+
+    Using a trained model, binned spectrums will be converted into spectrum
+    vectors using a deep neural network. The MS2DeepScoreMonteCarlo similarity is then
+    the mean of n_ensemble x n_ensemble cosine similarity score between two spectrum
+    vectors.
+
+    Example code to calcualte MS2DeepScoreMonteCarlo similarities between query and
+    reference spectrums:
+
+    .. code-block:: python
+
+        from matchms import calculate_scores()
+        from matchms.importing import load_from_json
+        from ms2deepscore import MS2DeepScoreMonteCarlo
+        from ms2deepscore.models import load_model
+
+        # Import data
+        references = load_from_json("abc.json")
+        queries = load_from_json("xyz.json")
+
+        # Load pretrained model
+        model = load_model("model_file_123.hdf5")
+
+        similarity_measure = MS2DeepScoreMonteCarlo(model, n_ensembles=5)
+        # Calculate scores and get matchms.Scores object
+        scores = calculate_scores(references, queries, similarity_measure)
+
+
+    """
+    def __init__(self, model, n_ensembles: int = 10, progress_bar: bool = True):
+        """
+
+        Parameters
+        ----------
+        model:
+            Expected input is a SiameseModel that has been trained on
+            the desired set of spectra. The model contains the keras deep neural
+            network (model.model) as well as the used spectrum binner (model.spectrum_binner).
+        n_ensembles
+            Number of embeddings to create for every spectrum using Monte-Carlo Dropout.
+            n_ensembles will lead to n_ensembles^2 scores of which the mean and STD will
+            be taken.
+        progress_bar:
+            Set to True to monitor the embedding creating with a progress bar.
+            Default is False.
+        """
+        self.model = model
+        self.n_ensembles = n_ensembles
+        self.input_vector_dim = self.model.base.input_shape[1]  # TODO: later maybe also check against SpectrumBinner
+        self.output_vector_dim = self.model.base.output_shape[1]
+        self.progress_bar = progress_bar
+        self.partial_model = self._create_monte_carlo_encoder()
```

Comment:
Maybe be consistent in naming. I would suggest:
```python
self.monte_carlo_base = self._create_monte_carlo_base()
```

Reply:
Done